app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -248,13 +269,32 @@ export default function createRPCMethodTrackingMiddleware({
   ) {
     const { origin, method, params } = req;
 
-    const rateLimitType = RATE_LIMIT_MAP[method];
+    const isMultichainRequest = isMultichainRequestMethod(method);
+    // requestedThrough and eventCategory are currently redundant so we will want to
+    // disentangle them in the future
+    const requestedThrough = isMultichainRequest
+      ? MetaMetricsRequestedThrough.MultichainApi
+      : MetaMetricsRequestedThrough.EthereumProvider;
+
+    const eventCategory = isMultichainRequest
+      ? MetaMetricsEventCategory.MultichainApi
+      : MetaMetricsEventCategory.InpageProvider;
+
+    let invokedMethod = method;
+    let multichainApiRequestScope;
+    // if the request is through the multichain api, the method is nested in the params
+    if (method === MESSAGE_TYPE.WALLET_INVOKE_METHOD) {
+      invokedMethod = params?.request?.method;
+      multichainApiRequestScope = params?.scope;
+    }
+
+    const rateLimitType = RATE_LIMIT_MAP[invokedMethod];
 
     let isRateLimited;
     switch (rateLimitType) {
       case RATE_LIMIT_TYPES.TIMEOUT:
         isRateLimited =
-          typeof rateLimitTimeoutsByMethod[method] !== 'undefined';
+          typeof rateLimitTimeoutsByMethod[invokedMethod] !== 'undefined';
```

Comment:
i know this was `'undefined'` before, but why lol. and is this correct even

Reply:
Well this is just an object where [the timeout for that rate limited method is temporarily stored](https://github.com/MetaMask/metamask-extension/blob/8a492c9fbd263d0787b4f0f9aaa8637293cb3925/app/scripts/lib/createRPCMethodTrackingMiddleware.js#L436), so theoretically if its undefined it means its no longer rate limited on this execution?

Reply:
i mean it's exact value comparing to the string `'undefined'`, not the actual undefined value `undefined`